### PR TITLE
WeakAuras: Add a stage trigger for BigWigs

### PR DESCRIPTION
**Pull request by Justw8**
# Description

This change adds a bigwigs stage trigger which can be used for weak auras.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested

- [x] Changed the stages in bigwigs using the callback, and it then shows/hides the WA depending on the stage settings set in the trigger.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: WeakAuras/Prototypes.lua [hunk 1]
```diff
@@ -4768,6 +4768,31 @@ Private.event_prototypes = {
...
+    init = function(trigger)
+      WeakAuras.RegisterBigWigsCallback("BigWigs_SetStage");
+      return [[
+         local stage = WeakAuras.GetBigWigsStage()
+      ]]
+    end,
+    args = {
+      {
+        name = "stage",
+        init = "arg",
```
Comment (InfusOnWoW) [inline WeakAuras/Prototypes.lua:4787]: It's "better" to use init = "WeakAuras.GetBigWigsStage()" here.

File: WeakAuras/Prototypes.lua [hunk 2]
```diff
@@ -4768,6 +4768,31 @@ Private.event_prototypes = {
...
+        display = L["Stage"],
+        type = "number",
+        conditionType = "number",
```
Comment (InfusOnWoW) [inline WeakAuras/Prototypes.lua:4790]: Has this been tested? I think a store = true is missing to make conditions work.

File: WeakAuras/GenericTrigger.lua [hunk 3]
```diff
@@ -2964,6 +2965,10 @@ do
           WeakAuras.ScanEvents("BigWigs_StopBar", id)
         end
       end
+    elseif event == "BigWigs_SetStage" then
+      local addon, stage = ...
+      currentStage = stage
+      WeakAuras.ScanEvents("BigWigs_SetStage", ...)
```
Comment (InfusOnWoW) [inline WeakAuras/GenericTrigger.lua:2971]: Is this event also send on encounter end/reset? (I would assume not, and then it probably requires adjusting the code in the BigWigs_OnBossDisable above?

Comment (Justw8) [reply WeakAuras/GenericTrigger.lua:2971]: Yea, I discussed this with Funkeh and we probablly will reset the stage number on boss resets/wins. We don't do this rightnow.

File: WeakAuras/Prototypes.lua (hunk 2 above)
Comment (Justw8) [reply WeakAuras/Prototypes.lua:4790]: Yea, right now it seems to work fine. see: https://streamable.com/ui1mwu
I can try it with and without store = true and see what happens.

File: WeakAuras/GenericTrigger.lua (hunk 3 above)
Comment (Justw8) [reply WeakAuras/GenericTrigger.lua:2971]: to add: we do always set it to 1 on loading a boss module/engaging right now.

File: WeakAuras/Prototypes.lua (hunk 2 above)
Comment (Justw8) [reply WeakAuras/Prototypes.lua:4790]: I didn't see a difference in store with or without.

**Comment by Justw8** [comment]
You can test it with this: 
https://gist.github.com/Justw8/730270f176791d32f44b5f6144d85f84

**Comment by InfusOnWoW** [comment]
Thanks

**Pull request merged**